src/test/java/org/codehaus/plexus/components/io/resources/PlexusIoPlexusIoFileResourceTest.java

```diff
@@ -21,7 +21,7 @@ public void testRealSymlink()
         final File file = new File( "src/test/resources/symlinks/src/symDir" );
         PlexusIoResourceAttributes attrs = FileAttributes.uncached( file );
         assertTrue( attrs.isSymbolicLink() );
-        PlexusIoFileResource r = new PlexusIoFileResource( file, "symDir", attrs );
```

Comment:
If my understanding is correct this change does not keep the current behavior of `PlexusIoFileResource`. If this is the case it should be a separate change (possibly requiring major version update if it is breaking change) as it is not just performance improvement.

Reply:
I disagree.  The `PlexusIoFileResource` has no public constructor, so the only way it can be created is through `ResourceFactory.createResource` or other internal methods.
What happens is that I think this change restores the original idea : `PlexusIoSymlinkResource` instances are used to represent symlinks and `PlexusIoFileResource` for regular files.

Reply:
The problem with the initial commit, is that I missed the fact that attributes are handled in a very specific (and not necessarily intuitive way) for links : some attributes are retrieved from the link and some others from the target.
That's the reason why I've introduced the `getLink()` and `getTarget()` methods on the `PlexusIoSymlinkResource` which return `PlexusIoResource` instances that will more explicitly handle the link and the target.

Reply:
Thanks for the explanation. Now it makes sense.